core/quantity/src/Parser.ts

```diff
@@ -3,21 +3,49 @@
 * See LICENSE.md in the project root for license terms and full copyright notice.
 *--------------------------------------------------------------------------------------------*/
 import { QuantityConstants } from "./Constants";
-import { QuantityStatus } from "./Exception";
 import { Format } from "./Formatter/Format";
 import { FormatTraits, FormatType } from "./Formatter/FormatEnums";
 import { PotentialParseUnit, QuantityProps, UnitConversion, UnitConversionSpec, UnitProps, UnitsProvider } from "./Interfaces";
 import { ParserSpec } from "./ParserSpec";
 import { Quantity } from "./Quantity";
 
+/** Possible parser errors
+ * @alpha
+ */
+export enum ParseError {
+  UnableToGenerateParseTokens = 1,
+  NoValueOrUnitFoundInString,
+  UnitLabelSuppliedButNotMatched,
+  UnknownUnit,
+  UnableToConvertParseTokensToQuantity,
+  InvalidParserSpec,
+}
+
+/** Parse error result from [[Parser.parseToQuantityValue]] or [[Parser.parseToQuantityValue]].
+ * @alpha
+ */
+export interface ParseQuantityError {
+  /** Union discriminator for [[ParseKeyinResult]]. */
+  ok: false;
+  /** The specific error that occurred during parsing. */
+  error: ParseError;
+}
+
+/** Successful result from [[Parser.parseToQuantityValue]] or [[Parser.parseToQuantityValue]].
+ * @alpha
+ */
+export interface ParsedQuantity {
+  /** Union discriminator for [[ParseKeyinResult]]. */

```

Comment:
`/s/ParseKeyinResult/QuantityParseResult` here and above.

Reply:
Thanks, good catch.